test/unit/token/env-change-token-parser-test.js

```diff
@@ -61,10 +62,10 @@ describe('Env Change Token Parser', () => {
     const data = buffer.data;
     data.writeUInt16LE(data.length - 3, 1);
 
-    const parser = new Parser({ token() { } }, {}, {});
-    parser.write(data);
-    const token = parser.read();
+    const parser = StreamParser.parseTokens([data], {}, {});
 
-    assert.strictEqual(token, null);
+    for await (let token of parser) { // (4)
+      assert.strictEqual(token, null);
+    }
```

Comment:
I don't think this (and the addition of `yield null` in the `Parser.parseTokens` static method) is quite right. 😅 

The `.read()` in the NodeJS `Readable` stream API is a synchronous method that either returns the next element of the stream (which would be a token), or `null` if there is no data available yet.

There's no completely matching equivalent to this when using an async generator, but that's okay - we just need to replicate the meaning of this test correctly.

The old test was trying to make sure that if data that references an invalid envchange token type was parsed, **no token** was returned by the parser - the data written to the parser was basically skipped.

One way to translate this more closely to the async generator version would just be to fail immediately if a token is generated. Something like this:

```
ts
for await (let token of parser) {
  assert.fail('expected no token to be returned');
}
```